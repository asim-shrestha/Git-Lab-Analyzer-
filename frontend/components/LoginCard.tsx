import React from "react";
import axios, { AxiosResponse, AxiosError } from "axios";
import AppTextField from "./AppTextField";
import AppButton from "./AppButton";
import { useSnackbar } from 'notistack';
import CardLayout from "./CardLayout";
import {Box} from "@material-ui/core";

const LoginCard = () => {
    const { enqueueSnackbar } = useSnackbar();
    const [userName, setUserName] = React.useState<string>("");
    const [password, setPassword] = React.useState<string>("");

    const handleLogin = () => {
        enqueueSnackbar("Attempting login.", {variant: 'default',});
        axios
            .post(`http://localhost:8080/login`,{},{
                auth: {
                    username: userName,
                    password: password,
                }
            })
            .then((resp: AxiosResponse) => {
                setUserName("");
                setPassword("");
            })
            .catch((err: AxiosError) => {
                enqueueSnackbar(`Login failed: ${err.message}`, {variant: 'error',});
            })
    }

    return (
        <CardLayout size="sm">
            <AppTextField placeholder="Username" value={userName} onChange={(e) => setUserName(e.target.value)}/>
            <AppTextField placeholder="Password" value={password} onChange={(e) => setPassword(e.target.value)}/>
            <Box alignSelf="center">
                <AppButton color="primary" onClick={handleLogin}>Login</AppButton>
            </Box>
        </CardLayout>

    )
}


export default LoginCard;
