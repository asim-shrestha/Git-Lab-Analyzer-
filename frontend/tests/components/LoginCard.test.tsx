import React from 'react';
import LoginCard from '../../components/LoginCard';
import {render} from "@testing-library/react";

describe("LoginCard", () =>{
    const mockEnqueue = jest.spyOn(require('notistack'), "useSnackbar");
    let enqueueSnackbar = jest.fn();


    beforeAll(() =>{
        mockEnqueue.mockImplementation(() => {return {enqueueSnackbar}});
    })

    it("Snapshot LoginCard", () => {
        const { container } = render(
            <LoginCard />
        )
        expect(container).toMatchSnapshot();
    })

})