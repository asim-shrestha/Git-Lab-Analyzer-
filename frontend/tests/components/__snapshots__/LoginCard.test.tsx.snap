// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginCard Snapshot LoginCard 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-backgroundGradient-1"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <div
        class="MuiBox-root MuiBox-root-5"
        id="card-header"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI1IiBoZWlnaHQ9IjEyNSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <img
            alt="The Gitlab Logo"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
        >
          GitLab
          <br />
          Analyzer
        </h6>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiFilledInput-root makeStyles-root-6 MuiFilledInput-underline makeStyles-underline-8 MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input makeStyles-input-7"
            placeholder="Username"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiFilledInput-root makeStyles-root-6 MuiFilledInput-underline makeStyles-underline-8 MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input makeStyles-input-7"
            placeholder="Password"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-9"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-10 MuiButton-containedPrimary MuiButton-disableElevation"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
